postprocessor/src/main/java/com/azure/autorest/postprocessor/implementation/CodeFormatterUtil.java

```diff
@@ -96,8 +106,83 @@ private static Map<String, String> loadEclipseSettings() throws Exception {
      */
     private static String removeUnusedImports(String file) throws Exception {
-        return RemoveUnusedImports.removeUnusedImports(
-            ImportOrderer.reorderImports(file, JavaFormatterOptions.Style.GOOGLE));
+        CompilationUnit compilationUnit = StaticJavaParser.parse(file);
+        com.github.javaparser.ast.NodeList<ImportDeclaration> imports = compilationUnit.getImports();
+
+        // Nothing to clean up.
+        if (imports.isEmpty()) {
+            return file;
+        }
+
+        // Package declaration could be null.
+        PackageDeclaration packageDeclaration = compilationUnit.getPackageDeclaration().orElse(null);
+        String packageName = packageDeclaration != null ? packageDeclaration.getNameAsString() : null;
+
+        // Collect all names used in the file that aren't associated with the package or imports.
+        Set<String> types = compilationUnit.stream()
+            .filter(node -> node instanceof NodeWithIdentifier || node instanceof NodeWithName
+                || node instanceof NodeWithSimpleName)
+            .filter(node -> !node.isDescendantOf(packageDeclaration) && !(node instanceof PackageDeclaration))
+            .filter(node -> imports.stream().noneMatch(node::isDescendantOf) && !(node instanceof ImportDeclaration))
+            .map(node -> {
+                if (node instanceof NodeWithIdentifier) {
+                    return ((NodeWithIdentifier<?>) node).getIdentifier();
+                } else if (node instanceof NodeWithName) {
+                    return ((NodeWithName<?>) node).getNameAsString();
+                } else {
+                    return ((NodeWithSimpleName<?>) node).getNameAsString();
+                }
+            })
+            .collect(Collectors.toSet());
+
+        // Collect all the types used in the Javadoc comments.
+        compilationUnit.getAllComments().stream()
+            .filter(comment -> comment instanceof JavadocComment)
+            .map(comment -> (JavadocComment) comment)
+            .forEach(javadoc -> javadoc.parse().getBlockTags()
+                .forEach(tag -> tag.getName().ifPresent(types::add)));
+
+        // Get the list of imports that are unused.
+        Map<Integer, ImportDeclaration> importsToRemove = imports.stream().filter(importDeclaration -> {
+            String fullImportName = importDeclaration.getNameAsString();
+            if (Objects.equals(fullImportName, packageName)) {
+                return true;
+            }
+
+            String importType = importDeclaration.getName().getIdentifier();
+            return !types.contains(importType);
+        }).collect(Collectors.toMap(importDeclaration -> importDeclaration.getRange().get().begin.line,
+            importDeclaration -> importDeclaration));
+
+        // Get the list of duplicate imports.
+        imports.stream().collect(Collectors.groupingBy(ImportDeclaration::getNameAsString)).entrySet().stream()
```

Comment:
nit: it would be nice to format all functional method calls in separate lines:

```java
imports.stream()
       .collect(..)
       .entrySet()
       .stream()
       ...
```